### ISSUE TITLE
 sending anonymous usage data settings not respected 

### ISSUE THREAD

**Issue by icf20** [ipfs/ipfs-webui#1041]
i dint select to send any of the data yet it does 

`20:00:12​?begin_session=*&metrics=​--​*IP*3​xhr​https://countly.ipfs.io/i?begin_session=1&metrics=%7B%22_app_version%22%3A%222.4.5%22%2C%22_ua%22%3A%22Mozilla%2F5.0%20(Windows%20NT%206.1%3B%20Win64%3B%20x64)%20AppleWebKit%2F537.36%20(KHTML%2C%20like%20Gecko)%20Chrome%2F75.0.3731.0%20Safari%2F537.36%22%2C%22_resolution%22%3A%221920x1080%22%2C%22_density%22%3A1%2C%22_locale%22%3A%22en-US%22%7D&app_key=*snip*&device_id=*snip*&sdk_name=javascript_native_web&sdk_version=19.02.1&timestamp=1558361366821&hour=16&dow=1&consent=%7B%22sessions%22%3Atrue%2C%22events%22%3Atrue%2C%22views%22%3Atrue%2C%22location%22%3Atrue%2C%22crashes%22%3Afalse%7D`

![image](https://user-images.githubusercontent.com/17784575/58042308-9fc02480-7b3a-11e9-8ecd-263ce0122e32.png)

**Comment by olizilla** [comment]
That definitely shouldn't happen. To help us debug that, please could you add info about the version of webui that you're seeing that happen in? The truncated git revision should be displayed in the bottom left corner in the nav bar, like so

<img width="289" alt="Screenshot 2019-05-22 at 15 25 31" src="https://user-images.githubusercontent.com/58871/58182657-ed0cd500-7ca5-11e9-901e-74447adb1471.png">

**Comment by olizilla** [comment]
We've configured the countly-sdk with `require_consent: true`, which is supposed to disable all requests until specific consents are added: https://github.com/ipfs-shipyard/ipfs-webui/blob/83fbf45156f78c5fd90bb502e600d7f127c32749/src/bundles/analytics.js#L55-L74

so I may need to go digging in the countly sdk

**Comment by icf20** [comment]
> please could you add info about the version of webui that you're seeing that happen in?

Revision 83fbf45

**Comment by olizilla** [comment]
Can you share any more information to help debug this? Did you ever tick the "Help improve this app by sending anonymous usage data" box? Could you share the contents of you local indexedDB called `bundle-cache`? Does it have an "analytics" key in it? If so what's the value? 

You can find whats in your browsers indexeddb by opening the dev tools (in chrome) and going to the Application tab > Storage > IndexedDB > bundle-cache > bundle-cache

See below for an example
<img width="1392" alt="Screenshot 2019-07-17 at 12 14 34" src="https://user-images.githubusercontent.com/58871/61371527-ce355400-a88c-11e9-9f1d-d1c557d22d9f.png">

**Comment by icf20** [comment]
> 
> "{"version":1,"time":1558610721843,"data":{"lastEnabledAt":1558610721333,"lastDisabledAt":1558610721831,"consent":[]}}"
> --
> 

this ?

**Comment by djoek** [comment]
still present on Revision 1eef40c

i have nothing in indexedDB, however localStorage had a few variables.  One was called cly_queue which held a structure like the one @icf20 posted.  I deleted all cly_* variables,  but still getting calls to /?begin_session=1 ... etc

**Comment by h1z1** [comment]
Almost two years later and this is still enabled by default.  Why?  uBlock is blocking the requests but how many people know this is even happening?
![ipfsfail01](https://user-images.githubusercontent.com/17220053/114929352-337ff580-9e23-11eb-8b25-6db942d45020.png)

**Comment by Stebalien** [comment]
It's not enabled by default and it's unclear why you're seeing this. Can you reproduce this bug in a fresh browser profile?

**Comment by h1z1** [comment]
I disagree, it is enabled by default and yes, I can verify the profile has nothing to do with it.

**Comment by olizilla** [comment]
I am able to recreate the issue where clicking on "no thanks" triggers a request to countly.ipfs.io where it absolutely should not.
It's sending:
 
- the list of analytic catagories that are being declined 
```
%7B%22sessions%22%3Afalse%2C%22events%22%3Afalse%2C%22views%22%3Afalse%2C%22location%22%3Afalse%2C%22crashes%22%3Afalse%7D
```

- the `app_key`
```
&app_key=8fa213e6049bff23b08e5f5fbac89e7c27397612
```

- the `device_id` (most egregious)
```
&device_id=c2cb4124-1212-4ddf-bf26-9bab77e70121&
```

- the sdk_name and sdk_version
sdk_name=javascript_native_web&sdk_version=19.08

- `location` is nulled out
``
&location=
```

- and a timestamp, hour and day of week of the event
```
 &timestamp=1619194484798&hour=17&dow=5

**Comment by olizilla** [comment]
I am fixing the _"no thanks" triggers a request_ issue

**Comment by lidel** [comment]
@olizilla fixed this in #1768 and we shipped it in [v2.12.1](https://github.com/ipfs/ipfs-webui/releases/tag/v2.12.1)
